src/main/webapp/teacher/executionCourse/attendsSearch/attendsSearchApp.js

```diff
@@ -1,19 +1,35 @@
-
+//Copied from http://stackoverflow.com/questions/1584370/how-to-merge-two-arrays-in-javascript-and-de-duplicate-items
+Array.prototype.unique = function() {
+    var a = this.concat();
+    for(var i=0; i<a.length; ++i) {
+        for(var j=i+1; j<a.length; ++j) {
+            if(a[i].id === a[j].id)
+                a.splice(j--, 1);
+        }
+    }
+    return a;
+};
+//END OF COPY
 
 var app = angular.module("AttendsSearchApp",  [ 'ui.bootstrap']);
 
 
 app.filter('attendsFilter', function() {
     return function(attends, filters) {
-    	var workingStudentCheck = function(attendee){
-            for (var i = 0; i < filters.workingStudentTypes.length; i++) {
-                if (filters.workingStudentTypes[i].value
-                		&& attendee.workingStudent == filters.workingStudentTypes[i].working) {
+
+    	var studentStatuteTypeCheck = function(attendee) {
+    		if(filters.noStudentStatuteTypes.value && attendee.studentStatutes.length == 0) {
+    			return true;
+    		}
+    		for (var i = 0; i < filters.studentStatuteTypes.length; i++) {
+                if (filters.studentStatuteTypes[i].value && attendee.studentStatutes.map(function(statute) {
+                	return statute.type.id;
+                }).indexOf(filters.studentStatuteTypes[i].id) > -1) {
                     return true;
                 }
```

Comment:
this can be replace with [array.filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter)